ipa-core/src/protocol/context/semi_honest.rs

```diff
@@ -272,7 +261,7 @@ impl<'a, B: ShardBinding, F: ExtendableField> UpgradedContext<F> for Upgraded<'a
         T: Send,
         UpgradeContext<'a, Self, F>: UpgradeToMalicious<'a, T, M>,
     {
-        UpgradeContext::new(self.narrow(&UpgradeStep::UpgradeSemiHonest), NoRecord)
+        UpgradeContext::new(self.clone(), NoRecord)
```

Comment:
I don't really understand what's going on here, and I'm not sure this change is appropriate -- but it did resolve issues I had with compact gate after removing modulus conversion from OPRF IPA.

Reply:
iirc, it was implemented because an early version of compact gate had to use an ignorelist for steps that don't trigger communications and tracked active steps through `send`. #711 may shed some light on why it was done this way.

I could be wrong, but I believe compact gate uses `narrow` to track steps now, so it may be the reason why you're not getting panic. If you could share the errors you're getting from modulus conversion, we can dig deeper into this. For now it seems that it should just work

you may want to fix the comment too (lines 254..258)

Reply:
The errors were "error: ipa_macros::step expects an enum with variants that match the steps in steps.txt" and "error[E0277]: the trait bound `compact::Compact: StepNarrow<UpgradeStep>` is not satisfied" (i.e. the usual errors when a step is not exercised in the steps.txt generation).

Part of why I was unsure what to do here, is that these methods were not being exercised -- so I wasn't sure there would be test coverage for whatever change I made.

But with the benefit of your comment and some more thinking, I have a proposal here that I'm more confident in. I'm going to open a separate PR for it (to be considered in advance of this one).

Reply:
#1030. After this PR is merged / rebased to include that, it will need to stub these impls based on `cfg(descriptive-gate)`, because it removes the last place we invoke share upgrades in the protocol (modulus conversion). They'll later need to come back when we add MAC-based malicious security for PRF evaluation and shuffle.